src/main/cpp/auton/drivePrimitives/DriveStopDelay.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright 2025 Lake Orion Robotics FIRST Team 302
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+// OR OTHER DEALINGS IN THE SOFTWARE.
+//====================================================================================================================================================
+
+// C++ Includes
+#include <memory>
+#include <string>
+
+// FRC includes
+#include "frc/kinematics/ChassisSpeeds.h"
+#include "frc/Timer.h"
+
+// Team 302 includes
+#include "auton/drivePrimitives/DriveStopDelay.h"
+#include "auton/drivePrimitives/IPrimitive.h"
+#include "auton/PrimitiveParams.h"
+#include "chassis/definitions/ChassisConfigMgr.h"
+#include "chassis/definitions/ChassisConfig.h"
+#include "chassis/ChassisMovement.h"
+#include "configs/MechanismConfig.h"
+#include "configs/MechanismConfigMgr.h"
+#include "utils/logging/Logger.h"
+
+// Third Party Includes
+
+using namespace std;
+using namespace frc;
+
+//========================================================================================================
+/// @class  DriveStop
+/// @brief  This is an auton primitive that causes the chassis to not drive
+//========================================================================================================
+
+/// @brief constructor that creates/initializes the object
+DriveStopDelay::DriveStopDelay() : IPrimitive(),
+                                   m_maxTime(units::time::second_t(0.0)),
+                                   m_currentTime(0.0),
+                                   m_chassis(nullptr),
+                                   m_timer(make_unique<Timer>()),
+                                   m_autonSelector(AutonSelector::GetInstance())
+{
+    auto config = ChassisConfigMgr::GetInstance()->GetCurrentConfig();
+    m_chassis = config != nullptr ? config->GetSwerveChassis() : nullptr;
+
+    // get reference to notemanager in drivestop to check for state
```

Comment:
don't think this primitive needs to know anything about mechanisms; if we want to control mechanisms, we should use zones or DriveStop